tests/compat.sh

```diff
@@ -534,6 +534,15 @@ add_mbedtls_ciphersuites()
     esac
 }
 
+# o_check_ciphersuite STANDARD_CIPHER_SUITE
+o_check_ciphersuite()
+{
+    if [ "${1#*ECDH_ECDSA*}" != "$1" ] && \
+       [ "X${O_SUPPORT_ECDH}" = "XNO" ]; then
+            SKIP_NEXT="YES"
+    fi
```

Comment:
There are `ECDH_RSA` cipher suites as well. We want to skip all `*ECDH_*` cipher suites.

Also, `[ "${1#*ECDH_ECDSA*}" != "$1" ]` is a correct way to check whether `$1` contains the substring `ECDH_ECDSA`, but it would be clearer to use `case`. Also, the `X` prefix in comparisons is not useful since `$O_SUPPORT_ECDH` is non-empty and can't look like an operator — and even if it could, all modern and even most antique shells handle three-word conditionals correctly. So we could simply write
```suggestion
    if [ "${O_SUPPORT_ECDH}" = "NO" ]; then
        case "$1" in
            *ECDH_*) SKIP_NEXT="YES"
        esac
    fi
```